lib/ansible/modules/cloud/azure/azure_rm_sqldatabase.py

```diff
@@ -0,0 +1,483 @@
+
+extends_documentation_fragment:
+    - azure
+    - azure_tags
+
+author:
+    - "Zim Kalinowski (@zikalino)"
+
+'''
+
+EXAMPLES = '''
+  - name: Create (or update) SQL Database
+    azure_rm_sqldatabase:
+      resource_group: resource_group_name
+      server_name: server_name
+      name: database_name
+      tags: tags
+      location: location
+      collation: collation
+      create_mode: create_mode
+      source_database_id: source_database_id
+      source_database_deletion_date: source_database_deletion_date
+      restore_point_in_time: restore_point_in_time
+      recovery_services_recovery_point_resource_id: recovery_services_recovery_point_resource_id
+      edition: edition
+      max_size_bytes: max_size_bytes
+      requested_service_objective_id: requested_service_objective_id
+      requested_service_objective_name: requested_service_objective_name
+      elastic_pool_name: elastic_pool_name
+      read_scale: read_scale
+      sample_name: sample_name
+      zone_redundant: zone_redundant
+'''
+
+RETURN = '''
+id:
+    description:
+        - Resource ID.
+    returned: always
+    type: str
+    sample: id
+database_id:
+    description:
+        - The ID of the database.
+    returned: always
+    type: str
+    sample: database_id
+status:
+    description:
+        - The status of the database.
+    returned: always
+    type: str
+    sample: status
+'''
+
+from ansible.module_utils.azure_rm_common import AzureRMModuleBase
+
+try:
+    from msrestazure.azure_exceptions import CloudError
+    from msrestazure.azure_operation import AzureOperationPoller
+    from azure.mgmt.sql import SqlManagementClient
+    from msrest.serialization import Model
+except ImportError:
+    # This is handled in azure_rm_common
+    pass
+
+
+class Actions:
+    NoAction, Create, Update, Delete = range(4)
+
+
+class AzureRMDatabases(AzureRMModuleBase):
+    """Configuration class for an Azure RM SQL Database resource"""
+
+    def __init__(self):
+        self.module_arg_spec = dict(
+            resource_group=dict(
+                type='str',
+                required=True
+            ),
+            server_name=dict(
+                type='str',
+                required=True
+            ),
+            name=dict(
+                type='str',
+                required=True
+            ),
+            tags=dict(
+                type='dict',
+                required=False
+            ),
+            location=dict(
+                type='str',
+                required=False
+            ),
+            collation=dict(
+                type='str',
+                required=False
+            ),
+            create_mode=dict(
+                type='str',
+                required=False
+            ),
+            source_database_id=dict(
+                type='str',
+                required=False
+            ),
+            source_database_deletion_date=dict(
+                type='datetime',
```

Comment:
Not a valid type, see https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/basic.py#L855-L868 for actual options.